clients/src/main/java/org/apache/kafka/clients/admin/ListConsumerGroupOffsetsResult.java

```diff
@@ -17,33 +17,72 @@
 
 package org.apache.kafka.clients.admin;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.clients.admin.internals.CoordinatorKey;
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.KafkaFuture;
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.annotation.InterfaceStability;
 
-import java.util.Map;
-
 /**
- * The result of the {@link Admin#listConsumerGroupOffsets(String)} call.
+ * The result of the {@link Admin#listConsumerGroupOffsets(Map)} and
+ * {@link Admin#listConsumerGroupOffsets(String)} call.
  * <p>
  * The API of this class is evolving, see {@link Admin} for details.
  */
 @InterfaceStability.Evolving
 public class ListConsumerGroupOffsetsResult {
 
-    final KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> future;
+    final Map<CoordinatorKey, KafkaFuture<Map<TopicPartition, OffsetAndMetadata>>> futures;
 
-    ListConsumerGroupOffsetsResult(KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> future) {
-        this.future = future;
+    ListConsumerGroupOffsetsResult(final Map<CoordinatorKey, KafkaFuture<Map<TopicPartition, OffsetAndMetadata>>> futures) {
+        this.futures = futures;
     }
 
     /**
      * Return a future which yields a map of topic partitions to OffsetAndMetadata objects.
      * If the group does not have a committed offset for this partition, the corresponding value in the returned map will be null.
      */
     public KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> partitionsToOffsetAndMetadata() {
-        return future;
+        if (futures.size() != 1) {
+            throw new IllegalStateException("Offsets from multiple consumer groups were requested. " +
+                    "Use groupIdsToPartitionsAndOffsetAndMetadata() instead to get all futures.");
+        }
+        return futures.values().iterator().next();
     }
 
+    /**
+     * Return a map of group ids to their corresponding futures that yield a map of topic partitions to
+     * OffsetAndMetadata objects. If the group doesn't have a committed offset for a specific
+     * partition, the corresponding value in the returned map for that group id will be null.
+     */
+    public Map<String, KafkaFuture<Map<TopicPartition, OffsetAndMetadata>>> groupIdsToPartitionsAndOffsetAndMetadata() {
```

Comment:
Yes, I was tempted to change this when going through the code, but didn't want to change the KIP at this point. And looking at the KIP now, looks like it was called `values()` before, so we seem to have changed the name somehow. Since we are changing the KIP for the previous comment anyway, let's just do `offsets(String group)`. We have `all()` to get the full map anyway. Will update the KIP and PR.

Reply:
Added partitionsToOffsetAndMetadata(String) to be compatible with the existing method partitionsToOffsetAndMetadata().